lib/pure/parseopt2.nim

```diff
@@ -112,10 +112,10 @@ proc next(p: var OptParser) =
     p.key = token
     p.val = ""
 
-proc cmdLineRest*(p: OptParser): TaintedString {.rtl, extern: "npo2$1", deprecated.} =
-  ## Returns part of command line string that has not been parsed yet.
-  ## Do not use - does not correctly handle whitespace.
-  return p.cmd[p.pos..p.cmd.len-1].join(" ")
+proc cmdLineRest*(p: OptParser): TaintedString {.rtl, extern: "npo2$1".} =
+  ## Returns the part of command line string that has not been parsed yet,
+  ## properly quoted.
```

Comment:
No it is not "properly quoted", that is impossible because the quoting rules depend on the used shell on Unix.

Reply:
I can edit that comment (=> `properly quoted for most shells`) but do you have an example of such shell (ideally a commonly used shell, like bash/zsh) where it breaks?

Reply:
No. Maybe ``fish``?

Reply:
seems to work fine with fish, bash, sh, zsh, see added note in top-post